app/controllers/api/v1/users_controller.rb

```diff
@@ -31,6 +31,9 @@ def create
 
         if user.save
           session[:user_id] ||= user.id
+          session[:session_token] ||= user.session_token
+          session[:session_expiry] ||= user.session_expiry
```

Comment:
This block is dedicated for signing in, so it should be able to use the same logic for signin,
The keeping of already existed session data is error prone,
Example, because of the lack of reset to invalid sessions a user that got logged out because of stale session data may have an error when creating an account.